File: implementation/sipamato/src/test/java/ch/difty/sipamato/persistance/jooq/paper/slim/PaperSlimRecordMapperTest.java
package ch.difty.sipamato.persistance.jooq.paper.slim;

import static ch.difty.sipamato.persistance.jooq.paper.PaperRecordMapperTest.FIRST_AUTHOR;
import static ch.difty.sipamato.persistance.jooq.paper.PaperRecordMapperTest.ID;
import static ch.difty.sipamato.persistance.jooq.paper.PaperRecordMapperTest.PUBLICATION_YEAR;
import static ch.difty.sipamato.persistance.jooq.paper.PaperRecordMapperTest.TITLE;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

import org.jooq.RecordMapper;

import ch.difty.sipamato.db.tables.records.PaperRecord;
import ch.difty.sipamato.entity.projection.PaperSlim;
import ch.difty.sipamato.persistance.jooq.RecordMapperTest;

public class PaperSlimRecordMapperTest extends RecordMapperTest<PaperRecord, PaperSlim> {

    public static void entityFixtureWithoutIdFields(PaperSlim entity) {
        when(entity.getFirstAuthor()).thenReturn(FIRST_AUTHOR);
        when(entity.getTitle()).thenReturn(TITLE);
        when(entity.getPublicationYear()).thenReturn(PUBLICATION_YEAR);
    }

    @Override
    protected RecordMapper<PaperRecord, PaperSlim> getMapper() {
        return new PaperSlimRecordMapper();
    }

    @Override
    protected PaperRecord makeRecord() {
        PaperRecord record = new PaperRecord();
        record.setId(ID);
        record.setFirstAuthor(FIRST_AUTHOR);
        record.setTitle(TITLE);
        record.setPublicationYear(PUBLICATION_YEAR);

        record.setVersion(VERSION);
        record.setCreated(CREATED);
        record.setCreatedBy(CREATED_BY);
        record.setLastModified(LAST_MOD);
        record.setLastModifiedBy(LAST_MOD_BY);

        return record;
    }

    @Override
    protected void assertEntity(PaperSlim e) {
        assertThat(e.getId()).isEqualTo(ID.intValue());
        assertThat(e.getFirstAuthor()).isEqualTo(FIRST_AUTHOR);
        assertThat(e.getTitle()).isEqualTo(TITLE);
        assertThat(e.getPublicationYear()).isEqualTo(PUBLICATION_YEAR);
    }

}

File: core/core-persistence-jooq/src/test/kotlin/ch/difty/scipamato/core/persistence/paper/slim/PaperSlimRecordMapperTest.kt
package ch.difty.scipamato.core.persistence.paper.slim

import ch.difty.scipamato.core.db.tables.records.PaperRecord
import ch.difty.scipamato.core.entity.projection.PaperSlim
import ch.difty.scipamato.core.persistence.RecordMapperTest
import ch.difty.scipamato.core.persistence.paper.PaperRecordMapperTest
import io.mockk.every
import org.amshove.kluent.shouldBeEqualTo
import org.jooq.RecordMapper

internal class PaperSlimRecordMapperTest : RecordMapperTest<PaperRecord, PaperSlim>() {

    override val mapper: RecordMapper<PaperRecord, PaperSlim> = PaperSlimRecordMapper()

    override fun makeRecord(): PaperRecord =
        PaperRecord().apply {
            id = PaperRecordMapperTest.ID
            number = PaperRecordMapperTest.NUMBER
            firstAuthor = PaperRecordMapperTest.FIRST_AUTHOR
            title = PaperRecordMapperTest.TITLE
            publicationYear = PaperRecordMapperTest.PUBLICATION_YEAR
        }

    override fun setAuditFieldsIn(record: PaperRecord) {
        record.created = CREATED
        record.createdBy = CREATED_BY
        record.lastModified = LAST_MOD
        record.lastModifiedBy = LAST_MOD_BY
        record.version = VERSION
    }

    override fun assertEntity(entity: PaperSlim) {
        entity.id shouldBeEqualTo PaperRecordMapperTest.ID
        entity.number shouldBeEqualTo PaperRecordMapperTest.NUMBER
        entity.firstAuthor shouldBeEqualTo PaperRecordMapperTest.FIRST_AUTHOR
        entity.title shouldBeEqualTo PaperRecordMapperTest.TITLE
        entity.publicationYear shouldBeEqualTo PaperRecordMapperTest.PUBLICATION_YEAR
    }

    companion object {
        fun entityFixtureWithoutIdFields(entity: PaperSlim) {
            every { entity.number } returns PaperRecordMapperTest.NUMBER
            every { entity.firstAuthor } returns PaperRecordMapperTest.FIRST_AUTHOR
            every { entity.title } returns PaperRecordMapperTest.TITLE
            every { entity.publicationYear } returns PaperRecordMapperTest.PUBLICATION_YEAR

            auditFixtureFor(entity)
        }
    }
}
